### PR TITLE
Add mapboxId to all relevant classes

### PR DESCRIPTION
**Pull request by eugenes78**
### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [ ] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: MapboxSearch/autofill/src/main/java/com/mapbox/search/autofill/AddressAutofillResult.kt [hunk 1]
```diff
@@ -46,7 +58,9 @@ public class AddressAutofillResult internal constructor(
      * @suppress
      */
     override fun hashCode(): Int {
-        var result = suggestion.hashCode()
+        var result = id.hashCode()
+        result = 31 * result + mapboxId.hashCode()
```
Comment (mapbox-danny) [inline MapboxSearch/autofill/src/main/java/com/mapbox/search/autofill/AddressAutofillResult.kt:62]: Does this hash function still make sense with the introduction the mapboxId?
Should we consider simply using the mapboxId when one is available?

File: MapboxSearch/discover/src/main/java/com/mapbox/search/discover/DiscoverResult.kt [hunk 2]
```diff
@@ -70,7 +82,9 @@ public class DiscoverResult internal constructor(
      * @suppress
      */
     override fun hashCode(): Int {
-        var result = name.hashCode()
+        var result = id.hashCode()
+        result = 31 * result + mapboxId.hashCode()
+        result = 31 * result + name.hashCode()
         result = 31 * result + address.hashCode()
```
Comment (mapbox-danny) [inline MapboxSearch/discover/src/main/java/com/mapbox/search/discover/DiscoverResult.kt:88]: Same question here - does this function still make sense?

File: MapboxSearch/gradle/versions.gradle [hunk 3]
```diff
@@ -45,7 +45,7 @@ ext {
     common_sdk_version = '24.0.0'
     mapbox_base_version = '0.8.0'
 
-    search_native_version = '2.0.0-alpha.4'
+    search_native_version = '2.0.0-alpha.11'
```
Comment (mapbox-danny) [inline MapboxSearch/gradle/versions.gradle:48]: I believe @ahmedaly16 mentioned that we need to get to alpha.13

File: MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteResult.kt [hunk 4]
```diff
@@ -129,7 +141,9 @@ public class PlaceAutocompleteResult internal constructor(
      * @suppress
      */
     override fun hashCode(): Int {
-        var result = name.hashCode()
```
Comment (mapbox-danny) [inline MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteResult.kt:132]: See previous questions about hashCode function

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchResult.kt [hunk 5]
```diff
@@ -181,6 +188,7 @@ public class SearchResult internal constructor(
     override fun hashCode(): Int {
```
Comment (mapbox-danny) [inline MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchResult.kt:188]: One more... perhaps we can find a way to reference it once

**Comment by RingerJK** [comment]
could be added in code docs the difference between id and mapboxId?

File: MapboxSearch/autofill/src/main/java/com/mapbox/search/autofill/AddressAutofillResult.kt (hunk 1 above)
Comment (eugenes78) [reply MapboxSearch/autofill/src/main/java/com/mapbox/search/autofill/AddressAutofillResult.kt:62]: It must be some Java/Kotlin thing, I don't think we use it anywhere directly. Also, `mapboxId` may be empty.

File: MapboxSearch/gradle/versions.gradle (hunk 3 above)
Comment (eugenes78) [reply MapboxSearch/gradle/versions.gradle]: Done

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchResult.kt (hunk 5 above)
Comment (eugenes78) [reply MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchResult.kt:188]: Same

File: MapboxSearch/sdk/src/test/java/com/mapbox/search/tests_support/SdkTestDataFactory.kt [hunk 6]
```diff
@@ -217,22 +219,6 @@ internal fun createTestHistoryRecord(
     categories = categories,
 )
 
-internal fun createHistoryRecord(searchResult: SearchResult, timestamp: Long): HistoryRecord {
-    return HistoryRecord(
```
[the remark this answers is not in the document]
Comment (eugenes78) [reply MapboxSearch/sdk/src/test/java/com/mapbox/search/tests_support/SdkTestDataFactory.kt:221]: It's not used anywhere, just some cleanup.

File: MapboxSearch/discover/src/main/java/com/mapbox/search/discover/DiscoverResult.kt (hunk 2 above)
Comment (eugenes78) [reply MapboxSearch/discover/src/main/java/com/mapbox/search/discover/DiscoverResult.kt:88]: Answered in my comment above

File: MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteResult.kt (hunk 4 above)
Comment (eugenes78) [reply MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteResult.kt:132]: Same

**Review by globaltrouble — APPROVED** [review]

**Pull request merged**